.github/workflows/build_and_test.yml

```diff
@@ -60,4 +62,28 @@ jobs:
           file: lcov.info
           flags: unittests
           name: codecov-umbrella
-          verbose: true
\ No newline at end of file
+          verbose: true
+  
+  cross-build-and-test-for-aarch64-unknown-linux-gnu:
```

Comment:
Not a MUST, but what about trying to add this into the above matrix, using " if runner == linux", and then doing the extra step?

That would:
- avoid an extra checkout
- avoid an extra toolchain install (add the target when on linux)

making it all faster, plus

- we could try switching or removing the "if runner == macos" and then get coverage measured for "pi_hw" also...

Sorry for not thinking of that earlier.....

if it got problematic for some reason, we can do it like this though.